src/data/int/gcd.lean

```diff
@@ -126,3 +126,14 @@ begin
 end
 
 end int
+
+lemma pow_gcd_eq_one {M : Type*} [monoid M] (x : M) {m n : ℕ} (hm : x ^ m = 1) (hn : x ^ n = 1) :
+  x ^ m.gcd n = 1 :=
```

Comment:
An idea (possibly, not for this PR): if we prove the trivial lemma `x ^ m = 1 ↔ is_periodic_pt (* x) m 1`, then this lemma follows from `is_periodic_pt.gcd`. And `order_of x = minimal_period (* x) 1` should help with some lemmas in `order_of_element` (it should be easy to make `minimal_period` computable by adding a `[decidable_pred]` assumption).

Reply:
Sounds good, but indeed, let's do that in another PR